### PR TITLE
Bug 1986495: Add missing translation for deployment strategy

### PR DESCRIPTION
**Pull request by jerolimov**
**Fixes**: 
https://issues.redhat.com/browse/ODC-6210
https://bugzilla.redhat.com/show_bug.cgi?id=1986495

**Analysis / Root cause**: 
In the Edit Deployment and Edit DeploymentConfig form, the strategy type dropdown help text is not pseudolocalized.

**Solution Description**: 
Add translation, unit test for the util functions and add some missing types.

**Screen shots / Gifs for design review**: 

With pseudo translation before:

before

With pseudo translation after:

after

**Unit test coverage report**: 
Add small unit tests for new utils function:

```
 PASS  packages/dev-console/src/components/edit-deployment/deployment-strategy/utils/__tests__/deployment-strategy-utils.spec.ts
  getDeploymentStrategyItems
    ✓ returns items for kubernetes resources (3ms)
    ✓ returns items for kubernetes resources
    ✓ returns no items for kubernetes resources
  getDeploymentStrategyHelpText
    ✓ return helptext for all known resource and deployment strategy combinations (1ms)
```

**Test setup:**
1. Create a deployment
2. Edit deployment
3. Add `?pseudolocalization=true&lng=en` in the url

**Browser conformance**: 
- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

File: frontend/packages/dev-console/src/components/edit-deployment/deployment-strategy/utils/deployment-strategy-utils.ts [hunk 1]
```diff
@@ -1,5 +1,63 @@
...
-import { LifecycleAction } from './types';
+import { Resources } from '../../../import/import-types';
+import { DeploymentStrategyType, LifecycleAction } from './types';
+
+export const getDeploymentStrategyItems = (resourceType: Resources, t: TFunction) => {
+  switch (resourceType) {
+    case Resources.Kubernetes:
+      return {
+        [DeploymentStrategyType.recreateParams]: t('devconsole~Recreate'),
+        [DeploymentStrategyType.rollingUpdate]: t('devconsole~Rolling Update'),
```
Comment (vikram-raj) [inline frontend/packages/dev-console/src/components/edit-deployment/deployment-strategy/utils/deployment-strategy-utils.ts:10]:
```suggestion
        [DeploymentStrategyType.rollingUpdate]: t('devconsole~Rolling update'),
```

Comment (jerolimov) [reply frontend/packages/dev-console/src/components/edit-deployment/deployment-strategy/utils/deployment-strategy-utils.ts:10]: Hmm it was "Rolling Update" before. It's fine for me to change this, but the detail page shows this:


image


And this is the "Edit Deployment" page at the moment:


image


@divyanshiGupta and @rebeccaalpert Wdyt?

Comment (rebeccaalpert) [reply frontend/packages/dev-console/src/components/edit-deployment/deployment-strategy/utils/deployment-strategy-utils.ts:10]: What do you think @spadgett?

**Comment by jerolimov** [comment]
This added an issue that the form do not allows saving when ImageStreamTag is used (default import from git flow). Will check this asap.

/hold

**Comment by jerolimov** [comment]
Rebased this change so that the "Edit Deployment" form could be saved with an internal ImageStreamTag again. This issue was not introduced with this PR. It was added as part of #9461 and is already fixed in #9637.

/hold cancel

**Review by vikram-raj — COMMENTED** [review]
/lgtm

**Pull request merged**